### ISSUE TITLE
first example crashes

### ISSUE THREAD

**Issue by jrobbins-LiveData** [mosquito/aiohttp-asgi#8]
I ran the first `README.md` example exactly as-is, and it crashed with this traceback (on Ubuntu, but also crashes with similar error on Windows 10)

```
python example1.py
unhandled exception during asyncio.run() shutdown
task: <Task finished name='Task-2' coro=<_run_app() done, defined at /mnt/c/Users/jeff1/Documents/projects/git/aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py:289> exception=RuntimeError("Task <Task pending name='Task-2' coro=<_run_app() running at /mnt/c/Users/jeff1/Documents/projects/git/aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py:323> cb=[_run_until_complete_cb() at /home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/base_events.py:184]> got Future <Future pending> attached to a different loop")>
Traceback (most recent call last):
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py", line 516, in run_app
    loop.run_until_complete(main_task)
  File "/home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py", line 323, in _run_app
    await runner.setup()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_runner.py", line 279, in setup
    self._server = await self._make_server()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_runner.py", line 375, in _make_server
    await self._app.startup()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_app.py", line 417, in startup
    await self.on_startup.send(self)
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiosignal/__init__.py", line 36, in send
    await receiver(*args, **kwargs)  # type: ignore
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp_asgi/resource.py", line 337, in on_startup
    msg = await sends.get()
  File "/home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/queues.py", line 166, in get
    await getter
RuntimeError: Task <Task pending name='Task-2' coro=<_run_app() running at aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py:323> cb=[_run_until_complete_cb() at /home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/base_events.py:184]> got Future <Future pending> attached to a different loop
Traceback (most recent call last):
  File "aio-test-u/example1.py", line 42, in <module>
    web.run_app(aiohttp_app)
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py", line 516, in run_app
    loop.run_until_complete(main_task)
  File "/home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py", line 323, in _run_app
    await runner.setup()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_runner.py", line 279, in setup
    self._server = await self._make_server()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_runner.py", line 375, in _make_server
    await self._app.startup()
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web_app.py", line 417, in startup
    await self.on_startup.send(self)
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiosignal/__init__.py", line 36, in send
    await receiver(*args, **kwargs)  # type: ignore
  File "aio-test-u/.venv/lib/python3.9/site-packages/aiohttp_asgi/resource.py", line 337, in on_startup
    msg = await sends.get()
  File "/home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/queues.py", line 166, in get
    await getter
RuntimeError: Task <Task pending name='Task-2' coro=<_run_app() running at aio-test-u/.venv/lib/python3.9/site-packages/aiohttp/web.py:323> cb=[_run_until_complete_cb() at /home/jeff1/.pyenv/versions/3.9.13/lib/python3.9/asyncio/base_events.py:184]> got Future <Future pending> attached to a different loop
Task was destroyed but it is pending!
task: <Task pending name='Task-1' coro=<FastAPI.__call__() running at aio-test-u/.venv/lib/python3.9/site-packages/fastapi/applications.py:267>>
sys:1: RuntimeWarning: coroutine 'FastAPI.__call__' was never awaited
```

**Comment by jrobbins-LiveData** [comment]
Changing this code (`startup=True`) to (`startup=False`) prevents the crash.

This crashes
```
# [Optional]
asgi_resource.lifespan_mount(
    aiohttp_app,
    startup=True,
    # By default starlette didn't
    # handle "lifespan.shutdown"
    shutdown=False,
)
```

This does not crash, but presumably then doesn't do anything useful?
```
# [Optional]
asgi_resource.lifespan_mount(
    aiohttp_app,
    startup=False,
    # By default starlette didn't
    # handle "lifespan.shutdown"
    shutdown=False,
)
```

**Comment by jrobbins-LiveData** [comment]
The issue appears to be due to the fact that `lifespan_mount` instances an asyncio `Queue` before `aiohttp` has created the asyncio event loop.  

```
    def lifespan_mount(self, app: Application, startup=True, shutdown=False):
        receives = asyncio.Queue()      # type: asyncio.Queue
```

Since there is no event loop, `asyncio.Queue()` creates one(!)

```
    def __init__(self, maxsize=0, *, loop=None):
        if loop is None:
            self._loop = events.get_event_loop()
```

Which is not what aiohttp expects in its `web.run_app()`.

**Comment by jrobbins-LiveData** [comment]
Using a coroutine to `web.run_app()` seems to fix the problem.  See [this](https://docs.aiohttp.org/en/stable/web_advanced.html#passing-a-coroutine-into-run-app-and-gunicorn) doc which states 

> The coroutine based approach allows to perform async IO before making an app.

```
from aiohttp import web
from fastapi import FastAPI
from starlette.requests import Request as ASGIRequest

from aiohttp_asgi import ASGIResource

asgi_app = FastAPI()


@asgi_app.get("/asgi")
async def root(request: ASGIRequest):
    return {
        "message": "Hello World",
        "root_path": request.scope.get("root_path")
    }


async def app_factory():
    aiohttp_app = web.Application()

    # Create ASGIResource which handle
    # any request startswith "/asgi"
    asgi_resource = ASGIResource(
        asgi_app,
        root_path="/asgi"
    )

    # Register resource
    aiohttp_app.router.register_resource(
        asgi_resource
    )

    # [Optional]
    asgi_resource.lifespan_mount(
        aiohttp_app,
        startup=True,
        # By default starlette didn't
        # handle "lifespan.shutdown"
        shutdown=False,
    )

    return aiohttp_app

# Start the application
web.run_app(app_factory())
```

**Comment by mosquito** [comment]
Fixed in 0.5.0
